doc/esmvalcore/preprocessor.rst

```diff
@@ -138,344 +260,559 @@ losing generality. To mask ice out, `mask_landseaice` can be used:
+      regrid_preprocessor:
+        regrid:
+          target_grid: 2.5x2.5
+          lon_offset: True
+          lat_offset: True
+          scheme: nearest
+
+Regridding (interpolation, extrapolation) schemes
+-------------------------------------------------
+
+The schemes used for the interpolation and extrapolation operations needed by
+the horizontal regridding functionality directly map to their corresponding
+implementaions in Iris:
+
+* ``linear``: `Linear(extrapolation_mode='mask') <https://scitools.org.uk/iris/docs/latest/iris/iris/analysis.html#iris.analysis.Linear>`_.
+* ``linear_extrapolate``: `Linear(extrapolation_mode='extrapolate') <https://scitools.org.uk/iris/docs/latest/iris/iris/analysis.html#iris.analysis.Linear>`_.
+* ``nearest``: `Nearest(extrapolation_mode='mask') <https://scitools.org.uk/iris/docs/latest/iris/iris/analysis.html#iris.analysis.Nearest>`_.
+* ``area_weighted``: `AreaWeighted() <https://scitools.org.uk/iris/docs/latest/iris/iris/analysis.html#iris.analysis.AreaWeighted>`_.
+* ``unstructured_nearest``: `UnstructuredNearest() <https://scitools.org.uk/iris/docs/latest/iris/iris/analysis.html#iris.analysis.UnstructuredNearest>`_.
+
+See also :func:`esmvalcore.preprocessor.regrid`
+
+.. note::
+
+   For both vertical and horizontal regridding one can control the
+   extrapolation mode when defining the interpolation scheme. Controlling the
+   extrapolation mode allows us to avoid situations where extrapolating values
+   makes little physical sense (e.g. extrapolating beyond the last data
+   point). The extrapolation mode is controlled by the `extrapolation_mode`
+   keyword. For the available interpolation schemes available in Iris, the
+   extrapolation_mode keyword must be one of:
+
+        * ``extrapolate`` – the extrapolation points will be calculated by
+	  extending the gradient of the closest two points;
+        * ``error`` – a ``ValueError`` exception will be raised, notifying an
+	  attempt to extrapolate;
+        * ``nan`` – the extrapolation points will be be set to NaN;
+        * ``mask`` – the extrapolation points will always be masked, even if
+	  the source data is not a ``MaskedArray``; or
+        * ``nanmask`` – if the source data is a MaskedArray the extrapolation
+	  points will be masked, otherwise they will be set to NaN.
+
+.. note::
+
+   The regridding mechanism is (at the moment) done with fully realized data in
+   memory, so depending on how fine the target grid is, it may use a rather
+   large amount of memory. Empirically target grids of up to ``0.5x0.5``
+   degrees should not produce any memory-related issues, but be advised that
+   for resolutions of ``< 0.5`` degrees the regridding becomes very slow and
+   will use a lot of memory.
+
+
+.. _multi-model statistics:
+
+Multi-model statistics
+======================
+Computing multi-model statistics is an integral part of model analysis and
+evaluation: individual models display a variety of biases depedning on model
```

Comment:
Should be "depending" instead of "depedning"?

Reply:
:+1: